### PR TITLE
Extract a common interface out of GaneshCanvas (into GPUCanvas) that can also be used for the Vulkan canvas.

### PR DESCRIPTION
**Pull request by chinmaygarde**
There is still some work to do here on the GPURasterizer. The interaction between the platform view (which provides the FBO, surface, etc.)  and the GPUCanvas is super awkward. Cleaning that up next.

### CODE REVIEW

File: shell/gpu/gpu_canvas.h [hunk 1]
```diff
@@ -5,11 +5,29 @@
...
 #include "lib/ftl/macros.h"
+#include "lib/ftl/compiler_specific.h"
+#include "third_party/skia/include/core/SkCanvas.h"
 
 namespace shell {
 
-//
+class GPUCanvas {
+ public:
+  static std::unique_ptr<GPUCanvas> SelectPlatformCanvas(
```
Comment (abarth) [inline shell/gpu/gpu_canvas.h]: Why "select" instead of "create" ?

Comment (chinmaygarde) [inline shell/gpu/gpu_canvas.h]: I imagined there would be a variety of options on the platform and this method would "select" a suitable one. Create works too. Patching.

Comment (chinmaygarde) [inline shell/gpu/gpu_canvas.h]: Done.

**Comment by chinmaygarde** [comment]
@abarth: What do you think about making the platform view select an own an instance of the GPUCanvas? I think the ownership works better that way since there are platform specific resources the canvas depends (like the ANativeWindow* on Android or the CAEAGLLayer* on iOS). The platform can also better communicate resizing and context loss information to the canvas. I didn't changes any ownership rules in this patch but I think this will be less awkward.

Also, GPUCanvas is a bit of a misnomer since the object actually is a vendor of canvases. It just so happens that only a single canvas can be active at given time on the GL backend since it has to be tied to the surface. But in the Vulkan backend, the application can be preparing multiple canvases (one per image in the swapchain). How about renaming the object to `GPUCanvasVendor` or `GPUCanvasFactory`?

**Comment by abarth** [comment]
> What do you think about making the platform view select and own an instance of the GPUCanvas?

That seems fine.  I was originally trying to separate the ownership of the objects according to what thread they were used on, but that turns out not to work because of the underlying ANativeWindow/CAEAGLLayer lifetime.

> How about renaming the object to GPUCanvasVendor or GPUCanvasFactory?

Sure.  I think we've used the term "provider" elsewhere for similar classes.

**Comment by chinmaygarde** [comment]
> I think we've used the term "provider" elsewhere for similar classes.

Ok. Will use `provider` for consistency. Do you want me to add to this patch or send another PR?

**Comment by abarth** [comment]
More, smaller PRs are usually better if its not too much extra work.

**Comment by chinmaygarde** [comment]
Sounds good. May I land this?

**Comment by abarth** [comment]
Go for it!

**Comment by chinmaygarde** [comment]
Was waiting for the lgtm without realizing you used the GitHub approval instead :D

**Pull request merged**